opt/scopeserver/dataserver/utils/loom_file_handler.py

```diff
@@ -12,65 +16,38 @@
 class LoomFileHandler:
     def __init__(self):
         self.active_looms = {}
-        self.loom_dir = dfh.DataFileHandler.get_data_dir_path_by_file_type(file_type="Loom")
+        self.file_locks = c.defaultdict(lambda: threading.Lock())
+        self.loom_dir = Path(dfh.DataFileHandler.get_data_dir_path_by_file_type(file_type="Loom"))
 
-    def add_loom(self, partial_md5_hash: str, file_path: str, abs_file_path: str, loom_connection):
+    def add_loom(self, file_path: Path, abs_file_path: Path, loom_connection) -> Loom:
         loom = Loom(
-            partial_md5_hash=partial_md5_hash,
-            file_path=file_path,
-            abs_file_path=abs_file_path,
-            loom_connection=loom_connection,
-            loom_file_handler=self,
+            file_path=file_path, abs_file_path=abs_file_path, loom_connection=loom_connection, loom_file_handler=self,
         )
         self.active_looms[abs_file_path] = loom
         return loom
 
-    def load_loom_file(self, partial_md5_hash: str, file_path: str, abs_file_path: str, mode: str = "r"):
+    def load_loom_file(self, file_path: Path, abs_file_path: Path, mode: str = "r") -> Optional[Loom]:
         try:
             loom_connection = lp.connect(abs_file_path, mode=mode, validate=False)
+            return self.add_loom(file_path=file_path, abs_file_path=abs_file_path, loom_connection=loom_connection,)
         except KeyError as e:
             logger.error(e)
             os.remove(file_path)
             logger.warning(f"Deleting malformed loom {file_path}")
+            self.file_locks[abs_file_path].release()
```

Comment:
Why is this here? Seems like it's going to trigger a [RuntimeError](https://docs.python.org/3.8/library/threading.html#threading.Lock.release) when the context manager tries to release it.

Reply:
Yeah, you're right. I forgot that the below return would still allow the contect manager to rexit gracefully, releasing the lock itself.